Fix Deployment Dashboard UI, added Route macros for routes

argha-dot opened the pull request:
Fixes #45 
Added app routes for deployment, runner, workspace routes

argha-dot committed:
fixed issue with deployment UI,
code cleanup for `app_type` matches.

argha-dot committed:
added App Route macros for declaring the routes for workspace,
deployment and runner

rakshith-ravi commented on hosted-frontend/src/routes/mod.rs:
```diff
@@ -1,14 +1,43 @@
+	/// The Sorting order of the deployments
+	#[serde(skip_serializing_if = "Option::is_none")]
+	pub order: Option<SortDirection>,
+	/// Filter the deployments by date
```
This has not been fixed yet

argha-dot committed:
added documentation to some components

argha-dot committed:
added sidebar items to sidebar

argha-dot committed:
- moved app_route to hosted-frontend
 - added app routes for database, deployments, workspaces, and runners
 - added steam logs api route

rakshith-ravi committed:
Refactor code and fix deployment UI issue

rakshith-ravi committed:
Added leptosfmt formatting

rakshith-ravi commented:
Merging this for now. Any updates can be put in a separate PR
